### ISSUE TITLE
TypeError: Cannot read properties of undefined (reading 'taskIds')

### ISSUE THREAD

**Issue by iangelovCC** [johannesjo/super-productivity#3018]
### Steps to Reproduce

1. Press Shift + A to create a new task.
2. Error message pops up. 

Alternative recreation: 
1. Click on "time not tracked" banner; click on add task - window to select or new task shows up. 
2. Write title for a new task to be added
3. Click on "Add Task" button
4. Error shows up (though the task has been saved). 

### Console Output

zone.js:1061 Unhandled Promise rejection: TypeError: Cannot read properties of undefined (reading 'taskIds') ; Zone: <root> ; Task: Promise.then ; Value: Error: TypeError: Cannot read properties of undefined (reading 'taskIds')
    at n.<anonymous> (global-error-handler.class.ts:45:13)
    at Generator.next (<anonymous>)
    at d (global-error-handler.util.ts:225:38)
    at U.invoke (zone.js:372:26)
    at it.run (zone.js:134:43)
    at zone.js:1275:36
    at U.invokeTask (zone.js:406:31)
    at it.runTask (zone.js:178:47)
    at w (zone.js:585:35)
    at B.invokeTask [as invoke] (zone.js:491:21) Error: TypeError: Cannot read properties of undefined (reading 'taskIds')
    at n.<anonymous> (file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:2568710)
    at Generator.next (<anonymous>)
    at d (file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:2565966)
    at U.invoke (file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/polyfills.8db5726c6ae16cfe.js:1:6670)
    at it.run (file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/polyfills.8db5726c6ae16cfe.js:1:2010)
    at file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/polyfills.8db5726c6ae16cfe.js:1:16980
    at U.invokeTask (file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/polyfills.8db5726c6ae16cfe.js:1:7290)
    at it.runTask (file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/polyfills.8db5726c6ae16cfe.js:1:2646)
    at w (file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/polyfills.8db5726c6ae16cfe.js:1:9348)
    at B.invokeTask [as invoke] (file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/polyfills.8db5726c6ae16cfe.js:1:8379)
E.onUnhandledError @ zone.js:1061
Show 1 more frame
Show less
console.js:77 16:02:13.198 › Frontend Error: TypeError: Cannot read properties of undefined (reading 'taskIds')
    at project.reducer.ts:730:69
    at ngrx-store.mjs:1382:26
    at ngrx-store.mjs:351:20
    at ngrx-store.mjs:304:43
    at action-logger.reducer.ts:14:12
    at undo-task-delete.meta-reducer.ts:128:12
    at ngrx-store.mjs:339:20
    at v.hi [as accumulator] (ngrx-store.mjs:486:21)
    at v._tryNext (scan.js:49:27)
    at v._next (scan.js:42:25) TypeError: Cannot read properties of undefined (reading 'taskIds')
    at file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:319644
    at file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:4147255
    at file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:4135967
    at file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:4135037
    at file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:2773055
    at file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:2771608
    at file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:4135367
    at v.hi [as accumulator] (file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:4137771)
    at v._tryNext (file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:3828656)
    at v._next (file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:3828557)
(anonymous) @ console.js:77
Show 1 more frame
Show less
console.js:77 16:02:14.049 › Frontend Error Stack: taskIds (webpack:///src/app/features/project/store/project.reducer.ts:730:68)
function (webpack:///node_modules/@ngrx/store/fesm2015/ngrx-store.mjs:1382:25)
createFeatureReducerFactory (webpack:///node_modules/@ngrx/store/fesm2015/ngrx-store.mjs:351:19)
combination (webpack:///node_modules/@ngrx/store/fesm2015/ngrx-store.mjs:306:44)
reducer (webpack:///src/app/root-store/meta/action-logger.reducer.ts:14:11)
reducer (webpack:///src/app/root-store/meta/undo-task-delete.meta-reducer.ts:128:11)
Array.isArray (webpack:///node_modules/@ngrx/store/fesm2015/ngrx-store.mjs:339:19)
reduceState (webpack:///node_modules/@ngrx/store/fesm2015/ngrx-store.mjs:486:20)
_tryNext (webpack:///node_modules/rxjs/_esm2015/internal/operators/scan.js:49:26)
_next (webpack:///node_modules/rxjs/_esm2015/internal/operators/scan.js:42:24)
(anonymous) @ console.js:77
ot.<computed> @ zone.js:2405
invokeTask @ zone.js:406
runTask @ zone.js:178
invokeTask @ zone.js:487
B.invoke @ zone.js:476
Z.B.args.<computed> @ zone.js:2385
Show 7 more frames
Show less
persistence.service.ts:337 LOAD COMPLETE false

### Stacktrace
```
TypeError: Cannot read properties of undefined (reading 'taskIds')
    at file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:319644
    at file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:4147255
    at file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:4135967
    at file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:4135037
    at file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:2773055
    at file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:2771608
    at file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:4135367
    at v.hi [as accumulator] (file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:4137771)
    at v._tryNext (file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:3828656)
    at v._next (file:///C:/Program%20Files/WindowsApps/53707johannesjo.SuperProductivity_7.17.2.0_x64__ch45amy23cdv6/app/resources/app.asar/dist/main.3e399feb98f3f00c.js:1:3828557)
```

### Meta Info
META: SP7.17.2 Electron – en-US – Win32 – Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) superProductivity/7.17.2 Chrome/120.0.6099.109 Electron/28.1.0 Safari/537.36

### Actions Before Error
```
1707141479364: [FocusMode] Set focus session elapsed time 
1707141479459: [SP_ALL] Load(import) all data 
1707141479461: [SP_ALL] All Data was loaded 
1707141484865: [Layout] Toggle AddTaskBar 
1707141492348: [Layout] Hide AddTaskBar 
1707141500462: [Task] Update Task 
1707141500556: [Persistence] Save to DB 
1707141501175: [Task] Update Task 
1707141501176: [Persistence] Save to DB 
1707141503264: [Task] Update Task 
1707141503268: [Persistence] Save to DB 
1707141510751: [Task] Update Task 
1707141510775: [Persistence] Save to DB 
1707141512430: [Task] Update Task 
1707141512431: [Persistence] Save to DB 
1707141515514: [Task] Update Task 
1707141515515: [Persistence] Save to DB 
1707141522033: [Task] Update Task 
1707141522036: [Persistence] Save to DB 
1707141523800: [Task] SetSelectedTask 
1707141523801: [Layout] Hide Notes 
1707141670713: [Task] Update Task 
1707141670719: [Persistence] Save to DB 
1707141689581: [Task] Update Task 
1707141689588: [Persistence] Save to DB 
1707141691544: [Layout] Toggle AddTaskBar 
1707141715553: [Layout] Hide AddTaskBar (2) 
1707141733119: [Task][Issue] Add Task 
1707141733122: [Persistence] Save to DB (2) 
1707141733122: [Task] Move tasks to other project
```

**Comment by iangelovCC** [comment]
Updated submitted issue with "Alternative recreation" of the issue under Steps to recreate. 

Form my limited understanding of the program, the issue is linked to the creation of new tasks. 

Submitting a crash export:
[ANONYMIZED-super-productivity-crash-user-data-export.json](https://github.com/johannesjo/super-productivity/file[ANONYMIZED-super-productivity-crash-user-data-export.json](https://github.com/johannesjo/super-productivity/files/14166809/ANONYMIZED-super-productivity-crash-user-data-export.json)

**Comment by johannesjo** [comment]
Hey there! Thank you very much for reporting this!

It might be that for some reason a bit of your data got into an invalid state that I didn't account for and that I can't reproduce. It would be great if you could share a copy of your data to contact@super-productivity.com, so I can debug this on my computer (and please let me know here, if you send something, as those mails quite often land in the spam folder). You can create an export by navigating to "Settings" and then to "Import/Export" or you can use the "Export data" button from the error Dialog (there is also a button there which will anonymize the data before exporting).

**Comment by johannesjo** [comment]
The upload of your data didn't work for some reason. You can also send it to contact@super-productivity.com

**Comment by iangelovCC** [comment]
Hey Johannes, 

Thank you for your time and effort in resolving this bug! As per your suggestion, I have sent a message to the email address you shared with me titled "Super Productivity: GitHub Issue #3018 "TypeError: Cannot read properties of undefined (reading 'taskIds')"". 

If there's anything else I may assist with, please let me know! 

Best regards, 
iangelovCC

P.S.: I will try once more to upload a new anonymized report through GitHub. 

[ANONYMIZED-super-productivity-crash-user-data-export (1).json](https://github.com/johannesjo/super-productivity/files/14223046/ANONYMIZED-super-productivity-crash-user-data-export.1.json)

**Comment by johannesjo** [comment]
I am not able to reproduce this even with the data you send me. Likely exporting your data and re-importing it into the newest version (8.0.0) will fix the issue.

**Comment by Jagdfalke** [comment]
@iangelovCC can you confirm that the issue is fixed with the latest version (currently 8.0.1)?
